src/dune_sexp/lexer.mll

```diff
@@ -348,7 +348,7 @@ and template_variable = parse
       (* -2 to account for the "%{" *)
       let start = { start with pos_cnum = start.pos_cnum - 2 } in
       let loc = Loc.create ~start ~stop:(Lexing.lexeme_end_p lexbuf) in
-      Template.Pform { loc ; name ; payload }
+      Template.Pform { loc ; name ; payload = Option.map ~f:(Template.Pform.Payload.create ~loc) payload }
```

Comment:
I think this location is incorrect. It should exclude the length of the name and the `:` that separates the two.

Reply:
Alternatively, you can just get rid of the location inside the payload and ask the callers to provide it.

Reply:
Yeah, I think that would be better actually. You would be able to use `Payload.t` when generating pforms without providing fake locations then.

Reply:
Updated to not store a `Loc.t` inside the payload and added a function to compute the payload's loc based on the pform's loc.